containers/base/build.sh

```diff
@@ -13,7 +13,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Builds the Google Cloud DataLab base docker image
+# Builds the Google Cloud DataLab base docker image. Usage:
+#   build.sh [path_of_pydatalab_dir]
+# If [path_of_pydatalab_dir] is provided, it will copy the content of that dir into image.
+# Otherwise, it will get the pydatalab by "git clone" from pydatalab repo.
 
 # Build the docker image
+if [ -n "$1" ]; then rsync -avp $1 py; else mkdir -p py; fi
```

Comment:
If you go with the "ADD py* ..." suggestion, then this line would simply become:

    if [ -n "$1" ]; then rsync -avp "$1" pydatalab; fi

Either way, though, make sure that the "$1" in the rsync command is quoted (in case the directory contains whitespace characters).